### PR TITLE
Support Cocoapods

### PR DESCRIPTION
**Pull request by alexmnps**
Thank you for your work on this library!

I've created a Podspec file so it can be included into projects that use Cocoapods for dependency management. I've tested it on one of the projects I'm working on, and the spec passes the lint validation (`pod lib lint`).

I improvised the extended `description` field, so you may want that changed.

### CODE REVIEW

**Comment by niw** [comment]
Hi @aceontech, thank you for your pull request.

I was thinking a little bit about CocoaPods and I am not strongly feeling it’s necessary to add podspec file to the repository. Because CocoaPods users can use this library without having a podspec file in this repository.

For example, add next simple podspac file as like the one in your patch in the project then add `pod 'KeyboardGuide', :podspec => 'path/to/KeyboardGuide.podspec'` in `Podfile`, then it can use this library, since it doesn't have any dependencies (other than UIKit.)
```
Pod::Spec.new do |spec|
  spec.name = "KeyboardGuide"
  spec.version = "0.2.0"
  spec.authors = ""
  spec.summary = "KeyboardGuide"
  spec.homepage = "https://github.com/niw/KeyboardGuide"
  spec.platform = :ios, "11.0"
  spec.source = {
    :git => "https://github.com/niw/KeyboardGuide.git", :tag => "#{spec.version}"
  }
  spec.source_files  = "Sources/KeyboardGuide/*.swift"
end
```

Also, since CocoaPods is managing its package version by the podspec file content, not git tags. Therefore adding podspec file to the repository causes a certain type of legacy and an unnecessary restriction to the project repository that need to sync git tag and podspec file content change.

Now, Swift Package Manager is integrated in Xcode and I think it is getting a de-facto package manager for Apple platform. CocoaPods is one of 3rd party package management tools, and there are a few others commonly used for Apple platform such as Carthage. If the project supports one of these 3rd party one, it will ends up with adding more and more management files to the project and these will be legacy at some point near future when Swift Package adaptation increased, so I am worrying of it as well.

I am not fully making decisions about CocoaPod support (or Carthage,) as long as it is commonly widely used in the community, and wondering how it is difficult to use Swift Package Manager alongside CocoaPods or others, or migrate to it.

Thoughts?

File: KeyboardGuide.podspec [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  spec.version      = "0.2.0"
+  spec.summary      = "A modern, real iOS keyboard system notifications handler framework that Just Works."
+  spec.description  = <<-DESC
+  A modern, real iOS keyboard system notifications handler framework that Just Works.
+  Based on years experience of iOS application development and various tests on each iOS version and device, it supports both Swift and Objective-C and works mostly reasonably on the latest 3 versions of iOS, which is iOS 11, 12 and iOS 13 now, and covers almost all iOS users.
+                   DESC
+
+  spec.homepage     = "https://github.com/niw/KeyboardGuide"
+  spec.license      = "MIT"
+  spec.author       = { "Yoshimasa Niwa" => "email@address.com" }
```
Comment (niw) [inline KeyboardGuide.podspec:12]: It can omit email, simply `spec.author = "Yoshimasa Niwa"` should be fine.

File: KeyboardGuide.podspec [hunk 2]
```diff
@@ -0,0 +1,21 @@
+Pod::Spec.new do |spec|
+  spec.name         = "KeyboardGuide"
+  spec.version      = "0.2.0"
+  spec.summary      = "A modern, real iOS keyboard system notifications handler framework that Just Works."
+  spec.description  = <<-DESC
+  A modern, real iOS keyboard system notifications handler framework that Just Works.
+  Based on years experience of iOS application development and various tests on each iOS version and device, it supports both Swift and Objective-C and works mostly reasonably on the latest 3 versions of iOS, which is iOS 11, 12 and iOS 13 now, and covers almost all iOS users.
+                   DESC
```
Comment (niw) [inline KeyboardGuide.podspec:8]: I think this `description` is not required, as you mentioned in the comment it's shorten version of content that users can see on GitHub repository.

File: KeyboardGuide.podspec [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+  spec.homepage     = "https://github.com/niw/KeyboardGuide"
+  spec.license      = "MIT"
+  spec.author       = { "Yoshimasa Niwa" => "email@address.com" }
+
+  spec.platform     = :ios, "11.0"
+
+  spec.source       = { :git => "https://github.com/niw/KeyboardGuide.git", :tag => "#{spec.version}" }
+  spec.source_files  = "Sources", "Sources/**/*.{h,m,swift}"
+  spec.exclude_files = "Sources/Exclude"
```
Comment (niw) [inline KeyboardGuide.podspec:18]: This `exclude_files` is not needed, and path seems irrelevant.

File: KeyboardGuide.podspec [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+                   DESC
+
+  spec.homepage     = "https://github.com/niw/KeyboardGuide"
+  spec.license      = "MIT"
+  spec.author       = { "Yoshimasa Niwa" => "email@address.com" }
+
+  spec.platform     = :ios, "11.0"
+
+  spec.source       = { :git => "https://github.com/niw/KeyboardGuide.git", :tag => "#{spec.version}" }
+  spec.source_files  = "Sources", "Sources/**/*.{h,m,swift}"
```
Comment (niw) [inline KeyboardGuide.podspec:17]: This should be `spec.source_files = "Sources/KeyboardGuide/*.swift"` a little bit explicitly, in case later adding test target or others, which shouldn't be a automatically part of `source_files`.

File: KeyboardGuide.podspec [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  spec.exclude_files = "Sources/Exclude"
+
+  spec.swift_version = '4.2'
```
Comment (niw) [inline KeyboardGuide.podspec:20]: This `swift_version` is not needed I think, also 4.2 is quite old.

**Comment by alexmnps** [comment]
Hi @niw!

Actually, I agree wholeheartedly with all of the points you made. I didn't know about the `:podspec` trick, so I'll be using that instead. With that in place, there is really no need to include the podspec. I will be moving my projects to SwiftPM soon anyway. 

Thanks!
